core/src/test/java/kafka/testkit/ClusterHarness.java

```diff
@@ -0,0 +1,15 @@
+package kafka.testkit;
+
+import kafka.network.SocketServer;
+import org.apache.kafka.common.network.ListenerName;
+
+import java.util.Collection;
+
+public interface ClusterHarness {
```

Comment:
This is injected into the generated test invocations at either the method or class level. This is intended to expose the common aspects of both cluster types.